### PR TITLE
[change] Exclude staff users from delete_unverified_users #353

### LINKED ISSUE #353: [change] Exclude staff users from delete_unverified_users
We should change this query:

https://github.com/openwisp/openwisp-radius/blob/b19de59455d40484674a2be870f99a2ee62568fe/openwisp_radius/management/commands/base/delete_unverified_users.py#L34-L38

To add `is_staff=False` to avoid deleting staff users in case they tried to register to their own service for testing purposes and didn't complete the verification.
We need to add a test / subtest and update the relevant documentation to make clear staff users are not deleted by this management command.

### PR DESCRIPTION
**Pull request by devkapilbansal**
Closes #353

### CODE REVIEW

File: docs/source/user/management_commands.rst [hunk 1]
```diff
@@ -92,6 +92,7 @@ This command deletes unverified users that have been registered for
 more than specified duration and have no associated radius session.
 This feature is needed to delete users who have registered but never
 completed the verification process.
+Staff users will be excluded from the deletion process.
```
Comment (nemesisdesign) [inline docs/source/user/management_commands.rst:95]:
```suggestion
**Staff users will not be deleted by this management command.**
```

File: openwisp_radius/tests/test_commands.py [hunk 2]
```diff
@@ -247,6 +247,26 @@ def _create_old_users():
...
+            )
+            user.is_staff = True
+            user.save()
+            self.assertEqual(User.objects.count(), 4)
+            call_command(
+                'delete_unverified_users',
+            )
+            self.assertEqual(User.objects.count(), 1)
+            self.assertEqual(
+                RadiusAccounting.objects.filter(username=opts['username']).exists(),
```
Comment (nemesisdesign) [inline openwisp_radius/tests/test_commands.py:266]: here I would change `opt['username']` to `user.username` to make this subtest block easier to read.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]

File: openwisp_radius/tests/test_commands.py [hunk 3]
```diff
@@ -247,6 +247,26 @@ def _create_old_users():
...
+            self.assertEqual(User.objects.count(), 4)
+            call_command(
+                'delete_unverified_users',
+            )
+            self.assertEqual(User.objects.count(), 1)
+            self.assertEqual(
+                RadiusAccounting.objects.filter(username=user.username).exists(),
+                True,
+            )
+
```
Comment (pandafy) [inline openwisp_radius/tests/test_commands.py:269]:
```suggestion
        with self.subTest('Staff users should not be deleted'):
            _create_old_users()
            user = self._create_user(
                date_joined=now() - timedelta(days=3), is_staff=True
            )
            RegisteredUser.objects.create(
                user=user,
                method='email',
                is_verified=False,
            )
            self.assertEqual(User.objects.count(), 4)
            call_command(
                'delete_unverified_users',
            )
            self.assertEqual(User.objects.count(), 1)
            self.assertEqual(
                User.objects.filter(username=user.username, is_staff=True).exists(),
                True,
            )
```

File: openwisp_radius/tests/test_commands.py [hunk 4]
```diff
@@ -247,6 +247,26 @@ def _create_old_users():
...
+            user.save()
+            self.assertEqual(User.objects.count(), 4)
+            call_command(
+                'delete_unverified_users',
+            )
+            self.assertEqual(User.objects.count(), 1)
+            self.assertEqual(
+                RadiusAccounting.objects.filter(username=user.username).exists(),
+                True,
+            )
```
Comment (pandafy) [inline openwisp_radius/tests/test_commands.py:268]: I believe this is added here by mistake.

**Review by pandafy — CHANGES_REQUESTED** [review]

File: docs/source/user/management_commands.rst [hunk 5]
```diff
@@ -92,6 +92,7 @@ This command deletes unverified users that have been registered for
 more than specified duration and have no associated radius session.
 This feature is needed to delete users who have registered but never
 completed the verification process.
+**Staff users will not be deleted from this management command.**
```
Comment (nemesisdesign) [inline docs/source/user/management_commands.rst:95]:
```suggestion
**Staff users will not be deleted by this management command.**
```

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**